internal/mode/static/handler_test.go

```diff
@@ -132,6 +147,68 @@ var _ = Describe("eventHandler", func() {
 		})
 	})
 
+	When("receiving control plane configuration updates", func() {
+		cfg := func(level nkgAPI.ControllerLogLevel) *nkgAPI.NginxGateway {
+			return &nkgAPI.NginxGateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      configName,
+				},
+				Spec: nkgAPI.NginxGatewaySpec{
+					Logging: &nkgAPI.Logging{
+						Level: helpers.GetPointer(level),
+					},
+				},
+			}
+		}
+
+		expStatuses := func(cond conditions.Condition) status.Statuses {
+			return status.Statuses{
+				NginxGatewayStatus: status.NginxGatewayStatus{
+					NSName: types.NamespacedName{
+						Namespace: namespace,
+						Name:      configName,
+					},
+					Conditions:         []conditions.Condition{cond},
+					ObservedGeneration: 0,
+				},
+			}
+		}
+
+		It("handles a valid config", func() {
```

Comment:
in the tests below, could we also test that the log level was set to the correct level?